packages/http-oracle-svc/test/unit/index.test.ts

```
@@ -29,31 +30,243 @@
  ******/
 
 "use strict";
-
+import fs from "fs";
+import {randomUUID} from "crypto";
 import {Service} from "../../src/service";
 
-describe("application X unit tests", () => {
+const partyType:string = "bank";
+const partyId:string = randomUUID();
+const fspId:string = "12";
+const url:string = `http://localhost:3031/participants/${partyType}/${partyId}?currency=UGX`;
+// jest.setTimeout(600000)
+
+const filePath = process.env["ORACLE_DB_FILE_PATH"];
+const defaultHeaders = new Headers();
+defaultHeaders.append("Content-Type", "application/json");
+
+
+describe("http-oracle-svc unit tests", () => {
 	beforeAll(async () => {
-		await Service.start();
+		if(filePath && fs.existsSync(filePath)){
+			/**  delete the db.json to start the server with no db.json **/
+			// Arrange
+			fs.unlinkSync(filePath);
+
+			// Act
+			const result = Service.start();
+
+			// Assert
+			await expect(result).resolves
+			await Service.stop();
+
+			/**  populate some data in the db.json to start the server with some data in the db.json **/
 
+			// Arrange
+			fs.writeFileSync(
+				filePath,
+				`{"associations": [{"partyId": "0b09ab3a-9a96-4f28-9672-1c9e4516a3fb","partyType": "bank","currency": "EUR","fspId": "12"},{"partyId": "0b09ab3a-9a96-4f28-9672-1c9e4516a3fb","partyType": "bank","currency": "EUR","fspId": "12"}]}`
+			);
+
+			// Act
+			await Service.start();
+
+			/**  populate empty data in the db.json to start the server with no data in the db.json **/
+			fs.writeFileSync(filePath,``);
+			await Service.stop();
+		}
+		await Service.start();
 	});
 
 	afterAll(async () => {
-		await Service.stop();
-
+		// await Service.stop();
 	});
 
 	/* Happy path tests */
-
 	test("create test oracle record", async () => {
-		// Test item goes here
-		// use axios or even better, nod's builtin fetch()
+		// Arrange
```

Hi @elijah0kello the test's name should follow the other ones on the rest of the packages.
You should be descriptive of its intentions. So for example:
"create test oracle record with no currency" - > "should create test oracle record with no currency"
"test start server with no db.json" - > "should throw error if no db.json is present on server"
"test create oracle with missing fspId" -> "should throw error when creating a oracle association with a missing fspId"

The tests should express an intention so that other developers are immediately aware of what we are testing on that specific test

The naming would helps in case of problems to figure out which problem we are facing.

---

@elijah0kello please use this naming strategy on the tests, example: 
- instead of "test start server with no db.json" could be "should throw error db.json exists with wrong content format on server start"
- "delete test oracle record with missing fspId" could be "delete test oracle record with missing fspId should return 422"

---

Okay. Will do that and update the recommended changes in my next commit.